consensus/src/block_storage/sync_manager.rs

```diff
@@ -255,45 +257,53 @@ impl BlockStore {
             blocks
                 .iter()
                 .take(blocks.len() - 1)
                 .map(|block| block.quorum_cert().clone()),
         );
 
+        // TODO: Make sure removing this code is okay
+
         // check if highest_commit_cert comes from a fork
         // if so, we need to fetch it's block as well, to have a proof of commit.
-        if !blocks
-            .iter()
-            .any(|block| block.id() == highest_commit_cert.certified_block().id())
-        {
-            info!(
-                "Found forked QC {}, fetching it as well",
-                highest_commit_cert
-            );
-            let mut additional_blocks = retriever
-                .retrieve_block_for_qc(
-                    highest_commit_cert,
-                    1,
-                    highest_commit_cert.certified_block().id(),
-                )
-                .await?;
-
-            assert_eq!(additional_blocks.len(), 1);
-            let block = additional_blocks.pop().expect("blocks are empty");
-            assert_eq!(
-                block.id(),
-                highest_commit_cert.certified_block().id(),
-                "Expecting in the retrieval response, for commit certificate fork, first block should be {}, but got {}",
-                highest_commit_cert.certified_block().id(),
-                block.id(),
-            );
-
-            blocks.push(block);
-            quorum_certs.push(highest_commit_cert.clone());
-        }
+        // if !blocks
+        //     .iter()
+        //     .any(|block| block.id() == highest_commit_cert.certified_block().id())
+        // {
+        //     info!(
+        //         "Found forked QC {}, fetching it as well",
+        //         highest_commit_cert
+        //     );
+        //     let mut additional_blocks = retriever
+        //         .retrieve_block_for_qc(
+        //             &QuorumCert::new(
+        //                 highest_commit_cert.vote_data().clone(),
+        //                 highest_commit_cert.ledger_info().clone(),
+        //             ),
+        //             1,
+        //             highest_commit_cert.certified_block().id(),
+        //         )
+        //         .await?;
+
+        //     assert_eq!(additional_blocks.len(), 1);
+        //     let block = additional_blocks.pop().expect("blocks are empty");
+        //     assert_eq!(
+        //         block.id(),
+        //         highest_commit_cert.certified_block().id(),
+        //         "Expecting in the retrieval response, for commit certificate fork, first block should be {}, but got {}",
+        //         highest_commit_cert.certified_block().id(),
+        //         block.id(),
+        //     );
+
+        //     blocks.push(block);
+        //     quorum_certs.push(QuorumCert::new(
+        //         highest_commit_cert.vote_data().clone(),
+        //         highest_commit_cert.ledger_info().clone(),
+        //     ));
+        // }
```

Comment:
having commented blocks are mainly for draft, we don't leave it like this in code

Reply:
I think this is an older commit. The latest commit doesn't have this block commented.

Reply:
As I introduced the order_vote_enable flag, this block will be executed when order_vote_enabled flag is set to false.